client.go

```diff
@@ -75,6 +94,9 @@ type Services interface {
 
 	// QueryInstances returns the list of available instances for a given service
 	QueryInstances(service string, opts QueryOptions) ([]ServiceInstance, error)
+
+	// WatchInstances returns a watch on instances updates for a given service
+	WatchInstances(service string, opts QueryOptions) (watch.Watch, error)
```

Comment:
Wondering if we should've named the package `xwatch` like we do with all the other top level `m3x` packages?  cc @mmihic @xichen2020